File: MyDailyBeat/MyDailyBeat/EVCJobsDetailsViewController.m
//
//  EVCJobsDetailsViewController.m
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 12/28/15.
//  Copyright © 2015 eVerveCorp. All rights reserved.
//

#import "EVCJobsDetailsViewController.h"
#import "EVCCommonMethods.h"
#import "API.h"

@interface EVCJobsDetailsViewController ()

@end

@implementation EVCJobsDetailsViewController

- (id) initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil andJob:(NSDictionary *) job {
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        self.jobEntry = job;
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIImage* image3 = [EVCCommonMethods imageWithImage:[UIImage imageNamed:@"hamburger-icon-white"] scaledToSize:CGSizeMake(30, 30)];
    CGRect frameimg = CGRectMake(0, 0, image3.size.width, image3.size.height);
    UIButton *someButton = [[UIButton alloc] initWithFrame:frameimg];
    [someButton setBackgroundImage:image3 forState:UIControlStateNormal];
    [someButton addTarget:self action:@selector(showMenu)
         forControlEvents:UIControlEventTouchUpInside];
    [someButton setShowsTouchWhenHighlighted:YES];
    
    UIBarButtonItem *menuButton =[[UIBarButtonItem alloc] initWithCustomView:someButton];
    
    self.tabBarController.navigationItem.rightBarButtonItem = menuButton;
    
    UIImage* image4 = [EVCCommonMethods imageWithImage:[UIImage imageNamed:@"profile-icon-white"] scaledToSize:CGSizeMake(30, 30)];
    CGRect frameimg2 = CGRectMake(0, 0, image4.size.width, image4.size.height);
    UIButton *someButton2 = [[UIButton alloc] initWithFrame:frameimg2];
    [someButton2 setBackgroundImage:image4 forState:UIControlStateNormal];
    [someButton2 addTarget:self action:@selector(showProfile)
          forControlEvents:UIControlEventTouchUpInside];
    [someButton2 setShowsTouchWhenHighlighted:YES];
    
    UIBarButtonItem *profileButton =[[UIBarButtonItem alloc] initWithCustomView:someButton2];
    self.tabBarController.navigationItem.leftBarButtonItem = profileButton;

    // Do any additional setup after loading the view from its nib.
    self.jobtitleLabel.text = [self.jobEntry objectForKey:@"jobtitle"];
    self.companyLabel.text = [self.jobEntry objectForKey:@"company"];
    self.locLabel.text = [self.jobEntry objectForKey:@"formattedLocationFull"];
    self.snippetLabel.text = [self.jobEntry objectForKey:@"snippet"];
    self.urlTextView.text = [self.jobEntry objectForKey:@"url"];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void) showMenu {
    [self.sideMenuViewController presentRightMenuViewController];
}

- (void) showProfile {
    [self.sideMenuViewController presentLeftMenuViewController];
}




@end

File: MDB_Swift/MyDailyBeat/MyDailyBeat/src/EVCJobsDetailsViewController.swift

//
//  EVCJobsDetailsViewController.swift
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 12/28/15.
//  Copyright © 2015 eVerveCorp. All rights reserved.
//
import UIKit
import RESideMenu
import API
import SwiftyJSON
class EVCJobsDetailsViewController: UIViewController {
    var jobEntry = JSON("")
    @IBOutlet var jobtitleLabel: UILabel!
    @IBOutlet var companyLabel: UILabel!
    @IBOutlet var locLabel: UILabel!
    @IBOutlet var snippetLabel: UILabel!
    @IBOutlet var urlTextView: UITextView!


    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view from its nib.
        self.jobtitleLabel.text = self.jobEntry["jobtitle"].stringValue
        self.companyLabel.text = self.jobEntry["company"].stringValue
        self.locLabel.text = self.jobEntry["formattedLocationFull"].stringValue
        self.snippetLabel.text = self.jobEntry["snippet"].stringValue
        self.urlTextView.text = self.jobEntry["url"].stringValue
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    
}
import API
